### PR TITLE
Implement keyless verification

### PR DESCRIPTION
**Pull request by flavio**
#### Summary

Implement verification of objects signed in keyless mode.

#### Ticket Link

Fixes https://github.com/sigstore/sigstore-rs/issues/11

### CODE REVIEW

File: src/cosign/signature_layers.rs [hunk 1]
```diff
@@ -41,210 +38,254 @@ use crate::{crypto::CosignVerificationKey, simple_signing::SimpleSigning};
...
+            None => None,
+        };
+        Ok(certificate_key)
+    }
+
+    /// Checks whether the SignatureLayer's can be verified using the given `verification_key`.
+    ///
+    /// When no `verification_key` is given, the verification will be done using the Certificate
+    /// and the Bundle that are eventually part of the SignatureLayer. If none of them is available,
+    /// the verification will fail.
```
Comment (lkatalin) [inline src/cosign/signature_layers.rs:175]: Is this why the verification key is now an optional parameter?

**Review by lkatalin — COMMENTED** [review]
I think this LGTM. Thanks for implementing this feature! I especially appreciate the thorough testing and documentation. Would it make sense in the documentation to point people to how they can retrieve the values for the Rekor pubkey (for the public instance) and Fulcio cert? I realized I am not sure of that.

File: src/cosign/signature_layers.rs (hunk 1 above)
Comment (flavio) [reply src/cosign/signature_layers.rs:175]: Yes, that's correct.

**Comment by flavio** [comment]
> Would it make sense in the documentation to point people to how they can retrieve the values for the Rekor pubkey (for the public instance) and Fulcio cert? I realized I am not sure of that.

My personal goal is to provide an API inside of this library to handle that. Unfortunately this is currently blocked by the Sigstore TUF repository containing "broken" (as in, not following the strict rules of Rust) data.
This is tracked here: https://github.com/sigstore/sigstore-rs/issues/9

The good news is that @asraa just got a fix merged (see https://github.com/sigstore/root-signing/pull/52). I think I can provide this API and then we can tag a new release of sigstore-rs which will have both this code, and the one that can fetch rekor+fulcio data via TUF

**Comment by lkatalin** [comment]
> > Would it make sense in the documentation to point people to how they can retrieve the values for the Rekor pubkey (for the public instance) and Fulcio cert? I realized I am not sure of that.
> 
> My personal goal is to provide an API inside of this library to handle that. Unfortunately this is currently blocked by the Sigstore TUF repository containing "broken" (as in, not following the strict rules of Rust) data. This is tracked here: #9
> 
> The good news is that @asraa just got a fix merged (see [sigstore/root-signing#52](https://github.com/sigstore/root-signing/pull/52)). I think I can provide this API and then we can tag a new release of sigstore-rs which will have both this code, and the one that can fetch rekor+fulcio data via TUF

Ah, I see! That's a great plan. Thanks @flavio

**Review by lkatalin — APPROVED** [review]

**Pull request merged**